sync/task_pool.go

```diff
@@ -160,6 +164,10 @@ func (p *TaskPool) run(id int, q chan task) error {
 
 // add task
 func (p *TaskPool) AddTask(t task) {
+	if t == nil {
+		return
```

Comment:
maybe there should be a panic when the @t  nil? @wongoo

Reply:
agree